File: DPM/dicty-genome-simulator.py
#Script to generate simulated genome based on Codon usaage frquency data
#author: prakhar gaur
#date: Tue July 07 IST 2015


import argparse

#argument parsing
parser = argparse.ArgumentParser()
parser.add_argument('lenghtofGenome', metavar='L', default='3400000', help='sequence lenght of the simulated genome in nucleotides, in base pairs (b.p.), default 3400000 b.p.', type=int)
parser.add_argument('-f', '--condonusageTable', help='text file with two columns, Col1 - Codon name, Col2 = Frequency of Codon per thousand, tab seprated, one codon entry per row', type=str)
args = parser.parse_args()

lenght = args.lenghtofGenome #default is assumed to be 3400000 b.p.
frequencyFile = args.condonusageTable


freqDict = {}

with open(frequencyFile, 'r') as inputFilehandle:
	for line in inputFilehandle:
		(key,value) = line.split()
		freqDict[key] = float(value)/1000
		
#print freqDict

with open("out1.txt", "w") as out1, open("out2.txt", "w") as out2:
	for (k,v) in freqDict.items():
		out1.write("%s\n" % k)
		out2.write("%s\n" % str(v))



	

File: DPM/dicty-genome-simulator.R
#Script to do the R part of Dicty genome simulation
#author : Prakhar Gaur
#Date : July 8 IST 2015

#Read data from File, probability values
probVal = read.table("probability-values-codon-usage.txt", header=FALSE)[,1]

#Read data from file, list of Codons (use stringsAsFactors=FALSE, otherwise codons are considered as Factors)
codon = read.table("codon-list.txt", header=FALSE, stringsAsFactors=FALSE)[,1]

#The sample() to generate the sequence, the function gives out one codon which means number of Seq length = nucleotides/3
dictyGenome <-sample(codon,1133333,replace=TRUE, probVal)

#Generate ad save frequency plot of Codons
#png("Condon-usage-freq-plot.png")
#plot(dictyGenome)
#dev.off()

#create string with fasta header
head <- c('>dictyGenome')

# Add fasta header to fasta file
write(head, "dicty-Synthetic-genome.fa")

#Append sequence to fasta file
write.table(paste(dictyGenome,collapse=""), "dicty-Synthetic-genome.fa", append=TRUE, col.name=FALSE, row.name=FALSE, quote=FALSE)

